.github/workflows/feature-branch-cleanup.yaml

```diff
@@ -0,0 +1,69 @@
+name: Feature Branch Cleanup
+
+on:
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  cleanup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check branch name
+        run: |
+          if [[ "${GITHUB_REF_NAME}" == "main" ]] || ! [[ "${GITHUB_REF_NAME}" =~ ^[0-9]+-.* ]]; then
+            echo "Feature branch cleanup can not be initiated on branch '$GITHUB_REF_NAME'" >> $GITHUB_STEP_SUMMARY
+            exit 0
+          fi
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.AWS_CLI_ROLE_ARN }}
+          role-session-name: js-assertion-cheat-sheet-feature-branch-cleanup
+          aws-region: ${{ secrets.AWS_REGION }}
+
+      - name: Setup Node.js
```

Comment:
I think this and the next steps are not necessary, as you don't execute any CDK command.